### ISSUE TITLE
E208 is too aggressive

### ISSUE THREAD

**Issue by greg-hellings** [ansible/ansible-lint#995]
##### Summary
When E208 is triggered, it should be sure that a file is being created. There are modules which only optionally create the file and, when not used in that form, then they should not require a mode be set. Such an example are ini_file, lineinfile, or blockinfile. These modules are frequently used in their default mode with `create: false`. Someone should not be required to set `mode: "0644"` or such in this situation.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible 2.9.12
  config file = /home/greg/.ansible.cfg
  configured module search path = ['/home/greg/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/greg/src/ansible_collections/devroles/system/.tox/lint_all/lib/python3.8/site-packages/ansible
  executable location = .tox/lint_all/bin/ansible
  python version = 3.8.5 (default, Aug 12 2020, 00:00:00) [GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]

ansible-lint --version
ansible-lint 4.3.1
```

- ansible installation method: one of source, pip, OS package
pip
- ansible-lint installation method: one of source, pip, OS package
pip

##### OS / ENVIRONMENT
Fedora 32

##### STEPS TO REPRODUCE
```yaml
- name: should not produce E208
  lineinfile:
    line: "# some content here"
    file: "{{ ansible_user_dir }}/.bashrc"

- name: SHOULD produce E208
  lineinfile:
    line: "# more content
    file: "{{ ansible_user_dir }}/.bash_profile"
    create: true
```
##### Desired Behaviour
ansible-lint should not error when it encounters a missing "mode" parameter for these modules, when `create: false`. Note that `create: false` is the default behavior of those modules

##### Actual Behaviour
ansible-lint throws E208 on both of the above steps
